### PR TITLE
use vcpkg to probe libmongoc package under windows

### PR DESCRIPTION
**Pull request by deepseeksss**
bacause of here are bugs in crate vcpkg and vcpkg's libmongoc, so should be test after new release, i will create issue and pr to them

### CODE REVIEW

**Comment by deepseeksss** [comment]
it should be test right now, just use vcpkg's git version

File: src/bsonc.rs [hunk 1]
```diff
@@ -28,10 +28,7 @@ impl Bsonc {
         try!(bson::encode_document(&mut buffer, document));
 
         let inner = unsafe {
-            bindings::bson_new_from_data(
-                buffer[..].as_ptr(),
-                buffer.len() as u64
-            )
+            bindings::bson_new_from_data(buffer[..].as_ptr(), (buffer.len() as u32).into())
```
Comment (thijsc) [inline src/bsonc.rs:31]: Why this change?

**Review by thijsc — COMMENTED** [review]
I think this should be ok. Could you explain why this is needed? I'm assuming it's some integration with the linking system of VS?

File: src/bsonc.rs (hunk 1 above)
Comment (deepseeksss) [reply src/bsonc.rs:31]: em....there is a problem with this change, close the pr first..
